internal/playlist/simulation.go

```diff
@@ -0,0 +1,192 @@
+package playlist
+
+import (
+	"context"
+	"log"
+	"sbercloud/internal/playback"
+	"sync"
+	"time"
+)
+
+type Simulation struct {
+	ctx context.Context
+
+	first      *track
+	last       *track
+	current    *track
+	trackCount int
```

Comment:
Self-implementation of double linked list is OK. Did you consider using a `container/list`?

Reply:
Export dll outside